### PR TITLE
Addon: helm-revision-pruner

### PR DESCRIPTION
**Pull request by rudimk**
Added the helm-revision-pruner chart as an addon.

### CODE REVIEW

File: addons/helm-revision-pruner/templates/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            {{- if .Values.global.image }}
+            image: "{{ .Values.global.image.repository }}:{{ .Values.global.image.tag }}"
+            {{- end }}
+            {{- else }}
+            image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+            {{- end }}
+            imagePullPolicy: {{ .Values.image.pullPolicy }}
+            {{- if .Values.global }}
+            {{- if .Values.global.image }}
+            {{- if and .Values.container.command (not (eq .Values.global.image.repository "public.ecr.aws/o1j4x7p4/hello-porter")) }}
```
Comment (ferozemohideen) [inline addons/helm-revision-pruner/templates/cronjob.yaml:69]: should all this logic about hello-porter image be here if the base image is `public.ecr.aws/porter-sre/helm-revision-pruner`?

**Comment by d-g-town** [comment]
One high-level question: why don't we use the porter job chart as the framework for this? If there are limitations in our chart, would be worth documenting/improving.

File: addons/helm-revision-pruner/templates/cronjob.yaml (hunk 1 above)
Comment (rudimk) [reply addons/helm-revision-pruner/templates/cronjob.yaml:69]: Yeah good point - I wanted to get the chart out asap so reused a lot of values from our job chart.

**Comment by rudimk** [comment]
> One high-level question: why don't we use the porter job chart as the framework for this? If there are limitations in our chart, would be worth documenting/improving.

So I thought of doing that at first, but the upgrader is written in a way where it's a lot easier to deploy a pre-made chart, rather than pull in the `job` chart, configure the values and push it. So I elected to build a new chart - but the values, the config, the templating functions - all taken from the `job` chart.

File: addons/helm-revision-pruner/templates/cronjob.yaml (hunk 1 above)
Comment (rudimk) [reply addons/helm-revision-pruner/templates/cronjob.yaml]: [1875111](https://github.com/porter-dev/porter-charts/pull/792/commits/1875111140baf2cc5a7e9c831d0d07d6f89cdf8a)

**Pull request merged**